scripts/ci-e2e.sh

```diff
@@ -70,8 +70,76 @@ source "${M3_DEV_ENV_PATH}/lib/ironic_tls_setup.sh"
 # image for live iso testing
 export LIVE_ISO_IMAGE="https://artifactory.nordix.org/artifactory/metal3/images/iso/minimal_linux_live-v2.iso"
 
+# Generate credentials
+BMO_OVERLAYS=(
+    "${REPO_ROOT}/test/e2e/data/bmo-deployment/overlays/release-0.4"
+    "${REPO_ROOT}/test/e2e/data/bmo-deployment/overlays/release-0.5"
+    "${REPO_ROOT}/test/e2e/data/bmo-deployment/overlays/release-0.6"
+    "${REPO_ROOT}/test/e2e/data/bmo-deployment/overlays/release-latest"
+)
+IRONIC_OVERLAYS=(
+    "${REPO_ROOT}/test/e2e/data/ironic-deployment/overlays/release-23.1"
+    "${REPO_ROOT}/test/e2e/data/ironic-deployment/overlays/release-24.0"
+    "${REPO_ROOT}/test/e2e/data/ironic-deployment/overlays/release-24.1"
+    "${REPO_ROOT}/test/e2e/data/ironic-deployment/overlays/release-latest"
+)
+
+# Create usernames and passwords and other files related to ironi basic auth if they
+# are missing
+if [[ "${IRONIC_BASIC_AUTH}" == "true" ]]; then
+    IRONIC_AUTH_DIR="${IRONIC_AUTH_DIR:-${IRONIC_DATA_DIR}/auth}"
+    mkdir -p "${IRONIC_AUTH_DIR}"
+
+    # If usernames and passwords are unset, read them from file or generate them
+    if [[ -z "${IRONIC_USERNAME:-}" ]]; then
+        if [[ ! -f "${IRONIC_AUTH_DIR}/ironic-username" ]]; then
+            IRONIC_USERNAME="$(uuid-gen)"
+            echo "${IRONIC_USERNAME}" > "${IRONIC_AUTH_DIR}/ironic-username"
+        else
+            IRONIC_USERNAME="$(cat "${IRONIC_AUTH_DIR}/ironic-username")"
+        fi
+    fi
+    if [[ -z "${IRONIC_PASSWORD:-}" ]]; then
+        if [ ! -f "${IRONIC_AUTH_DIR}/ironic-password" ]; then
+            IRONIC_PASSWORD="$(uuid-gen)"
+            echo "$IRONIC_PASSWORD" > "${IRONIC_AUTH_DIR}/ironic-password"
+        else
+            IRONIC_PASSWORD="$(cat "${IRONIC_AUTH_DIR}/ironic-password")"
+        fi
+    fi
+    IRONIC_INSPECTOR_USERNAME="${IRONIC_INSPECTOR_USERNAME:-${IRONIC_USERNAME}}"
+    IRONIC_INSPECTOR_PASSWORD="${IRONIC_INSPECTOR_PASSWORD:-${IRONIC_PASSWORD}}"
+
+    export IRONIC_USERNAME
+    export IRONIC_PASSWORD
+    export IRONIC_INSPECTOR_USERNAME
+    export IRONIC_INSPECTOR_PASSWORD
+fi
+
+for overlay in "${BMO_OVERLAYS[@]}"; do
```

Comment:
From L119 to L139 this is indented with 2 spaces while rest of the script is 4 spaces, please fix.